src/frequenz/sdk/actor/data_sourcing/common.py

```diff
@@ -0,0 +1,66 @@
+"""Common types for the DataSourcingActor.
+
+Copyright
+Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+
+License
+MIT
+"""
+
+from dataclasses import dataclass
+from datetime import datetime
+from enum import Enum
+from typing import Optional
+
+
+class ComponentMetric(Enum):
+    """An enum representing the various metrics available in the data pipeline."""
+
+    ACTIVE_POWER = "active_power"
+
+    CURRENT_PHASE_1 = "current_phase_1"
+    CURRENT_PHASE_2 = "current_phase_2"
+    CURRENT_PHASE_3 = "current_phase_3"
+
+    VOLTAGE_PHASE_1 = "voltage_phase_1"
+    VOLTAGE_PHASE_2 = "voltage_phase_2"
+    VOLTAGE_PHASE_3 = "voltage_phase_3"
+
+    SOC = "soc"
+    SOC_LOWER_BOUND = "soc_lower_bound"
+    SOC_UPPER_BOUND = "soc_upper_bound"
+    CAPACITY = "capacity"
+
+    POWER_LOWER_BOUND = "power_lower_bound"
+    POWER_UPPER_BOUND = "power_upper_bound"
+
+    ACTIVE_POWER_LOWER_BOUND = "active_power_lower_bound"
+    ACTIVE_POWER_UPPER_BOUND = "active_power_upper_bound"
+
+
+@dataclass
+class ComponentMetricRequest:
```

Comment:
This is first required by the `DataSourcingActor`, so I would move it to `data_sourcing.py`.

This way we can also completely get rid of the `common` module. Both `ComponentMetricRequest` and `ComponentMetricId` can be exposed to the user via `__init__`, so they can do `from frequenz.sdk.actor.data_sourcing import ComponentMetricRequest` for example, and they don't care where it is initially defined.

Reply:
Both `ComponentMetricId` and `ComponentMetricRequest` need to be accessible for other actors,  and will probably be accepted by the resampling actor too?  So I put it in the `frequenz/sdk/types.py`.  Does that sound ok?

Reply:
For me it doesn't sound like a general interface, I see it more owned by the data sourcing actor, even if other actors use it. It is completely coupled with how the data sourcing actor works and will have to be updated if the data sourcing actor *interface* changes, this is why I think they belong together.

Yes, probably the resampling actor will use it, but it can expose an alias, because eventually the resampling actor could have other parameters in the request (I was thinking that maybe, just **maybe**, we could want at some point to request a metric to be resampled with a particular resampling function after we talked about resampling a `xxx_max` metric for example). In that case it will have to define it's own request.

All that said, it is still a style issue I guess, so I won't oppose if you want to leave it like this. :)

Reply:
The `DataSourcingActor` will also be the first source of `Sample`s.  What do you think about just `frequenz/sdk/actor/data_sourcing/types.py` for all three?

(Also I had a typo in the path I mentioned earlier, that's now fixed)

Reply:
Haha, I thought about `Sample` but since I didn't have a good case for it I left it out in the hope that you won't notice :trollface: 

OK, now that you forced me to think about it harder, I found out why `Sample` is different for me: If we ever make the resampler agnostic of the data sourcing actor, `Sample` will be a more general concept, where other sourcing actors will also use `Sample`, but not `ComponentMetricId` and `ComponentMetricRequest`. This is why I think `Sample` is a more general data pipeline concept, not necessarily tied to the data sourcing actor, while the other too I don't see them being used outside of it (independently I mean, not to communicate with the data sourcing actor).

But since this is an hypothetical case for now, I would be fine to add it too to the data sourcing actor. At the end I think we'll probably need to re-arrange a thing of two as we advance more with the data pipeline and the structure is more clear.

About `types`, I'm not crazy about general modules like `common` and `types` but `types` is definitely better as it is more scoped than `common`. So personally I would still have `ComponentMetricId` and `ComponentMetricRequest` very close to the class that gives live to those classes (again, for me `Sample` is different), but `types` is fine too.